File: tests/sdk/nodejs/test/eks.test.mjs
import { test } from "node:test";
import assert from "node:assert/strict";
import {
  requireReady,
  signedFetch,
  uniquePrefix,
} from "../lib/helpers.mjs";

test("EKS CreateDescribeListDelete soft-skip when down", async (t) => {
  if (!(await requireReady(t))) return;
  const name = `jseks-${uniquePrefix()}`.replace(/_/g, "-").slice(0, 40);
  const createBody = Buffer.from(
    JSON.stringify({
      name,
      roleArn: "arn:aws:iam::000000000001:role/eks",
      version: "1.29",
      resourcesVpcConfig: {
        subnetIds: ["subnet-1"],
        securityGroupIds: [],
      },
    }),
  );
  const createResp = await signedFetch(
    "eks",
    "POST",
    "/clusters",
    createBody,
    "application/json",
  );
  assert.ok(createResp.status >= 200 && createResp.status < 300, createResp.status);
  t.after(async () => {
    await signedFetch("eks", "DELETE", `/clusters/${encodeURIComponent(name)}`);
  });

  const created = await createResp.json();
  const status = created?.cluster?.status || "";
  if (status !== "ACTIVE") {
    t.skip(`EKS live smoke skipped: status=${status}`);
    return;
  }
  assert.match(String(created?.cluster?.endpoint || ""), /noctaxris-eks-/);

  const desc = await signedFetch("eks", "GET", `/clusters/${encodeURIComponent(name)}`);
  assert.equal(desc.status, 200);
  const described = await desc.json();
  assert.equal(described?.cluster?.status, "ACTIVE");
  const list = await signedFetch("eks", "GET", "/clusters");
  const listText = await list.text();
  assert.equal(list.status, 200);
  assert.match(listText, new RegExp(name));

  const nodes = await signedFetch(
    "eks",
    "GET",
    `/clusters/${encodeURIComponent(name)}/node-groups`,
  );
  assert.equal(nodes.status, 200);

  const del = await signedFetch(
    "eks",
    "DELETE",
    `/clusters/${encodeURIComponent(name)}`,
  );
  assert.ok(del.status >= 200 && del.status < 300, del.status);
  const gone = await signedFetch("eks", "GET", `/clusters/${encodeURIComponent(name)}`);
  assert.notEqual(gone.status, 200);
});

File: tests/sdk/go/eks_test.go
package sdk_test

import (
	"encoding/json"
	"net/http"
	"strings"
	"testing"
)

func TestEKSCreateDescribeListDelete(t *testing.T) {
	requireReady(t)
	name := "goeks-" + uniquePrefix(t)
	if len(name) > 40 {
		name = name[:40]
	}
	createBody, err := json.Marshal(map[string]any{
		"name":    name,
		"roleArn": "arn:aws:iam::000000000001:role/eks",
		"version": "1.29",
		"resourcesVpcConfig": map[string]any{
			"subnetIds":        []string{"subnet-1"},
			"securityGroupIds": []string{},
		},
	})
	if err != nil {
		t.Fatal(err)
	}
	st, raw := signedHTTP(t, "eks", http.MethodPost, "/clusters", createBody, "application/json")
	if st < 200 || st >= 300 {
		t.Fatalf("CreateCluster status=%d body=%s", st, raw)
	}
	t.Cleanup(func() {
		_, _ = signedHTTP(t, "eks", http.MethodDelete, "/clusters/"+name, nil, "application/json")
	})

	var created map[string]any
	if err := json.Unmarshal(raw, &created); err != nil {
		t.Fatal(err)
	}
	cluster, _ := created["cluster"].(map[string]any)
	if cluster == nil {
		t.Fatalf("missing cluster: %s", raw)
	}
	status, _ := cluster["status"].(string)
	if status != "ACTIVE" {
		t.Skipf("EKS live smoke skipped: status=%s (expected ACTIVE metadata)", status)
	}
	ep, _ := cluster["endpoint"].(string)
	if !strings.Contains(ep, "noctaxris-eks-") {
		t.Fatalf("expected nested endpoint, got %q", ep)
	}

	st, descRaw := signedHTTP(t, "eks", http.MethodGet, "/clusters/"+name, nil, "application/json")
	if st != 200 {
		t.Fatalf("DescribeCluster status=%d body=%s", st, descRaw)
	}
	var described map[string]any
	if err := json.Unmarshal(descRaw, &described); err != nil {
		t.Fatal(err)
	}
	descCluster, _ := described["cluster"].(map[string]any)
	if descCluster == nil || descCluster["status"] != "ACTIVE" {
		t.Fatalf("DescribeCluster want ACTIVE got %v", described)
	}
	st, listRaw := signedHTTP(t, "eks", http.MethodGet, "/clusters", nil, "application/json")
	if st != 200 || !strings.Contains(string(listRaw), name) {
		t.Fatalf("ListClusters status=%d body=%s", st, listRaw)
	}
	st, ngRaw := signedHTTP(t, "eks", http.MethodGet, "/clusters/"+name+"/node-groups", nil, "application/json")
	if st != 200 {
		t.Fatalf("ListNodegroups status=%d body=%s", st, ngRaw)
	}
	st, delRaw := signedHTTP(t, "eks", http.MethodDelete, "/clusters/"+name, nil, "application/json")
	if st < 200 || st >= 300 {
		t.Fatalf("DeleteCluster status=%d body=%s", st, delRaw)
	}
	st, goneRaw := signedHTTP(t, "eks", http.MethodGet, "/clusters/"+name, nil, "application/json")
	if st == 200 {
		t.Fatalf("DescribeCluster after Delete still 200: %s", goneRaw)
	}
}
